pkg/chartutil/requirements.go

```diff
@@ -266,3 +268,128 @@ func ProcessRequirementsEnabled(c *chart.Chart, v *chart.Config) error {
 
 	return nil
 }
+
+// pathToMap creates a nested map given a YAML path in dot notation
+func pathToMap(path string, data map[string]interface{}) map[string]interface{} {
+	ap := strings.Split(path, ".")
+	n := []map[string]interface{}{}
```

Comment:
- [ ] You could use `make` here with a size the pre-allocate the backing array, given that it's always going to hold `len(ap)` entries.